SoftLayer/managers/dedicated_host.py

```diff
@@ -317,18 +318,57 @@ def _get_backend_router(self, locations, item):
                 if category['categoryCode'] == 'dedicated_host_disk':
                     disk_capacity = capacity['capacity']
 
+        for hardwareComponent in item['bundleItems']:
+            if hardwareComponent['keyName'].find("GPU") != -1:
+                hardwareComponentModel = hardwareComponent['hardwareGenericComponentModel']
+                hardwareGenericComponentModelId = hardwareComponentModel['id']
+                hardwareComponentType = hardwareComponentModel['hardwareComponentType']
+                hardwareComponentTypeKeyName = hardwareComponentType['keyName']
+
         if locations is not None:
             for location in locations:
                 if location['locationId'] is not None:
                     loc_id = location['locationId']
-                    host = {
-                        'cpuCount': cpu_count,
-                        'memoryCapacity': mem_capacity,
-                        'diskCapacity': disk_capacity,
-                        'datacenter': {
-                            'id': loc_id
+                    if item['keyName'].find("GPU") == -1:
+                        host = {
```

Comment:
this bit should be outside of the `if` statement since it always need to be set, and is the same for GPU/NO-GPU

Reply:
Done.